AppCenter/AppCenter/Internals/Ingestion/MSAppCenterIngestion.m

```diff
@@ -62,6 +64,13 @@ - (NSURLRequest *)createRequest:(NSObject *)data eTag:(NSString *)__unused eTag
   request.allHTTPHeaderFields = self.httpHeaders;
   [request setValue:self.appSecret forHTTPHeaderField:kMSHeaderAppSecretKey];
 
+  // Copy self.authToken into a local variable to avoid a race condition.
+  NSString *authTokenCopy = self.authToken;
+  if (authTokenCopy && ![authTokenCopy isEqualToString:@""]) {
```

Comment:
`if (!([authTokenCopy length] == 0))` should do the trick.

Reply:
Maybe `if ([authTokenCopy length] > 0)`?

Reply:
Even better Jae :)